tsdb/db.go

```diff
@@ -189,6 +192,11 @@ type Options struct {
 
 	// EnableSharding enables query sharding support in TSDB.
 	EnableSharding bool
+
+	// Delays the auto compactions start time by a random offset.
+	// This offset can be increased by up to one minute if the database does not commit too often.
+	EnableDelayedCompaction bool
+	CompactionDelay         time.Duration
```

Comment:
As far as I can see, someone calling into tsdb code can turn the feature on either by setting `EnableDelayedCompaction` true, or by setting `CompactionDelay` to a non-zero time.

This (a) isn't clearly documented and (b) seems redundant.

You could remove the boolean and call `rand` where it is currently set.
I'm not saying this is a big improvement, but maybe conceptually simpler.

Reply:
we need `db.head.chunkRange` for `generateCompactionDelay`, I don't think we should/can get its final value from `main.go`.

Before, I used a slightly similar approach: set the delay to `-1` if the flag is set, then regenerate it later: https://github.com/prometheus/prometheus/pull/12532#discussion_r1504098743, but it was confusing.

I tried to make the comments clearer, we can always improve this in the future.